presto-main/src/main/java/com/facebook/presto/operator/ExchangeOperator.java

```diff
@@ -47,7 +47,7 @@ public ExchangeOperatorFactory(int operatorId, PlanNodeId sourceId, Supplier<Exc
         {
             this.operatorId = operatorId;
             this.sourceId = sourceId;
-            this.exchangeClientSupplier = exchangeClientSupplier;
+            this.exchangeClientSupplier = new ExchangeClientSupplier(exchangeClientSupplier.get());
```

Comment:
Why wrap `exchangeClientSupplier` instead of modifying `ExchangeClientFactory` itself?